cluster/repository/identity_cluster.go

```diff
@@ -0,0 +1,132 @@
+type GormIdentityClusterRepository struct {
+	db *gorm.DB
+}
+
+// NewIdentityClusterRepository creates a new storage type.
+func NewIdentityClusterRepository(db *gorm.DB) IdentityClusterRepository {
+	return &GormIdentityClusterRepository{db: db}
+}
+
+// IdentityClusterRepository represents the storage interface.
+type IdentityClusterRepository interface {
+	Load(ctx context.Context, identityID, clusterID uuid.UUID) (*IdentityCluster, error)
+	ListClustersForIdentity(ctx context.Context, identityID uuid.UUID) ([]Cluster, error)
+	Create(ctx context.Context, u *IdentityCluster) error
+	Delete(ctx context.Context, identityID, clusterID uuid.UUID) error
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *GormIdentityClusterRepository) TableName() string {
+	return "identity_cluster"
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m IdentityCluster) TableName() string {
+	return "identity_cluster"
+}
+
+// CRUD Functions
+
+// Load returns a single Identity Cluster as a Database Model
+func (m *GormIdentityClusterRepository) Load(ctx context.Context, identityID, clusterID uuid.UUID) (*IdentityCluster, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "identity_cluster", "load"}, time.Now())
+	var native IdentityCluster
+	err := m.db.Table(m.TableName()).Preload("Cluster").Where("identity_id = ? and cluster_id = ?", identityID, clusterID).Find(&native).Error
+	if err == gorm.ErrRecordNotFound {
+		return nil, errors.NewNotFoundErrorFromString(fmt.Sprintf("identity_cluster with identity ID %s and cluster ID %s not found", identityID, clusterID))
+	}
+	return &native, errs.WithStack(err)
+}
+
+// ListClustersForIdentity returns the list of all cluster for the identity
+func (m *GormIdentityClusterRepository) ListClustersForIdentity(ctx context.Context, identityID uuid.UUID) ([]Cluster, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "identity_cluster", "listClustersForIdentity"}, time.Now())
+	var rows []IdentityCluster
+
+	err := m.db.Table(m.TableName()).Preload("Cluster").Where("identity_id = ?", identityID).Find(&rows).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, errs.WithStack(err)
+	}
+	clusters := make([]Cluster, 0, len(rows))
+	for _, idCluster := range rows {
+		clusters = append(clusters, idCluster.Cluster)
+	}
+	return clusters, nil
+}
+
+// Create creates a new record.
+func (m *GormIdentityClusterRepository) Create(ctx context.Context, c *IdentityCluster) error {
+	defer goa.MeasureSince([]string{"goa", "db", "identity_cluster", "create"}, time.Now())
+	err := m.db.Create(c).Error
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"cluster_id":  c.ClusterID.String(),
+			"identity_id": c.IdentityID.String(),
+			"err":         err,
+		}, "unable to create the cluster")
```

Comment:
how about `"unable to create identity cluster"`?

Reply:
Done in https://github.com/fabric8-services/fabric8-cluster/pull/25/commits/c53acb82e55793ca31aa2cb944e9333f53881ae7